R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+    "user_password <- getPass::getPass('OMERO password: ')\n",
+    "server <- OMEROServer(host = 'outreach.openmicroscopy.org', username=user_name, password=user_password, port= as.integer(4064))\n",
+    "server <- connect(server)\n",
+    "paste('Successfully logged in as', server@user$getUserName())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and display an image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load an image from the example dataset 'R-dataset' (these images are part of the original 'CDK5RAP2-C' dataset).\n",
+    "\n",
+    "**Exercise:** Go to the image 'siControl_N20_Cep215_I_20110411_Mon-1509_0_SIR_PRJ.dv' in your 'R-dataset', find the 'Image ID' and copy/paste it below. You can use a different image of the dataset if you like, but I recommend that one because it nicely shows the toroid centrioles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imageId <- REPLACE_WITH_IMAGE_ID # An image from R-dataset\n",
+    "image <- loadObject(server, \"ImageData\", imageId)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load the pixel values of the second channel (the CDK5RAP2-C channel) and display the image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pixels <- getPixelValues(image, 1, 1, 2) # Second channel of the first z plane of the first time point\n",
+    "                                         # (the images only have one z/t plane anyway)\n",
+    "ebi <- EBImage::Image(data = pixels, colormode = 'Grayscale')\n",
+    "ebi <- normalize(ebi)\n",
+    "EBImage::display(ebi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Image segmentation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Visual"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "threshImg <- thresh(ebi, w=15, h=15, offset=0.1)\n",
+    "threshImg <- medianFilter(threshImg, size=3)\n",
+    "threshImg <- fillHull(threshImg)          \n",
+    "threshImg <- bwlabel(threshImg)\n",
+    "EBImage::display(colorLabels(threshImg))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Modify the parameters of the thresh function to get a feeling for how the parameters affect the segmentation, or even plug in other methods into the segmentation workflow to improve the result, see [EBImage documentation](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compute the features (measurements)"
```

Comment:
Oh, this is not explained at all - what features am I computing ? What functions am I using ?

Reply:
Well, the functions in the code block. And the features or measuerments are displayed as table when the code block ran. Does that really need more explanation? Can add a link to the computeFeatures documentation maybe.